activation/poetdb.go

```diff
@@ -0,0 +1,135 @@
+package activation
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	xdr "github.com/nullstyle/go-xdr/xdr3"
+	"github.com/spacemeshos/go-spacemesh/common"
+	"github.com/spacemeshos/go-spacemesh/database"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/types"
+	"github.com/spacemeshos/merkle-tree"
+	"github.com/spacemeshos/poet/hash"
+	"github.com/spacemeshos/poet/shared"
+	"github.com/spacemeshos/poet/verifier"
+	"github.com/spacemeshos/sha256-simd"
+)
+
+type PoetDb struct {
+	store database.Database
+	log   log.Log
+}
+
+func NewPoetDb(store database.Database, log log.Log) *PoetDb {
+	return &PoetDb{store: store, log: log}
+}
+
+func (db *PoetDb) ValidateAndStorePoetProof(proof types.PoetProof, poetId []byte, roundId uint64, signature []byte) error {
```

Comment:
IMO i think that poet id should have a type and not be of byte array, it is too generic and cannot act as a key in go map. also, printing byte arrays is not something desirable in our logs

Reply:
I can accept making it a byte **array** (it's currently a slice). This would make it more specific and usable as a map key.
Printing byte slices (and arrays) with `%x` is easy and prints them as hex values.